src/Symfony/Component/Mailer/Bridge/Azure/Transport/AzureApiTransport.php

```diff
@@ -0,0 +1,283 @@
+     */
+    private function getPayload(Email $email, Envelope $envelope): array
+    {
+        $addressStringifier = function (Address $address) {
+            $stringified = ['address' => $address->getAddress()];
+
+            if ($address->getName()) {
+                $stringified['displayName'] = $address->getName();
+            }
+
+            return $stringified;
+        };
+
+        $data = [
+            'content' => [
+                'html' => $email->getHtmlBody(),
+                'plainText' => $email->getTextBody(),
+                'subject' => $email->getSubject(),
+            ],
+            'recipients' => [
+                'to' => array_map($addressStringifier, $this->getRecipients($email, $envelope)),
+            ],
+            'senderAddress' => $envelope->getSender()->getAddress(),
+            'attachments' => $this->getMessageAttachments($email),
+            'userEngagementTrackingDisabled' => !$this->engagementTracking,
+            'headers' => empty($headers = $this->getMessageCustomHeaders($email)) ? null : $headers,
+            'importance' => $this->getPriorityLevel($email->getPriority()),
+        ];
+
+        if ($emails = array_map($addressStringifier, $email->getCc())) {
+            $data['recipients']['cc'] = $emails;
+        }
+
+        if ($emails = array_map($addressStringifier, $email->getBcc())) {
+            $data['recipients']['bcc'] = $emails;
+        }
+
+        if ($emails = array_map($addressStringifier, $email->getReplyTo())) {
+            $data['replyTo'] = $emails;
+        }
+
+        return $data;
+    }
+
+    /**
+     * List of attachments. Please note that the service limits the total size
+     * of an email request (which includes attachments) to 10 MB.
+     */
+    private function getMessageAttachments(Email $email): array
+    {
+        $attachments = [];
+        foreach ($email->getAttachments() as $attachment) {
+            $headers = $attachment->getPreparedHeaders();
+            $filename = $headers->getHeaderParameter('Content-Disposition', 'filename');
+            $disposition = $headers->getHeaderBody('Content-Disposition');
+
+            $att = [
+                'name' => $filename,
+                'contentInBase64' => base64_encode(str_replace("\r\n", '', $attachment->bodyToString())),
+                'contentType' => $headers->get('Content-Type')->getBody(),
+            ];
+
+            if ('inline' === $disposition) {
+                $att['content_id'] = $filename;
+            }
+
+            $attachments[] = $att;
+        }
+
+        return $attachments;
+    }
+
+    /**
+     * The communication domain host, for example my-acs-resource-name.communication.azure.com.
+     */
+    private function getAzureCSEndpoint(): string
+    {
+        return !empty($this->host) ? $this->host : str_replace('%resource_name_dot%', 'my-acs-resource-name' !== ($this->resourceName ?: 'my-acs-resource-name') ? $this->resourceName.'.' : '', self::HOST);
```

Comment:
I would throw an exception in the constructor if the resourceName ends with a dot, this would simplify this a lot and communicates clear to the user.

I would also change the const:
```diff
- private const HOST = '%resource_name_dot%communication.azure.com';
- private const HOST = '%s.communication.azure.com';
```

and use `sprintf()` over str_replace